src/libpriv/rpmostree-core.cxx

```diff
@@ -2486,7 +2487,8 @@ rpmostree_context_download (RpmOstreeContext *self,
         rpmostree_output_progress_percent_begin (&progress, "Downloading from '%s'",
                                                  dnf_repo_get_id (src));
 
-        target_dir = g_build_filename (dnf_repo_get_location (src), "/packages/", NULL);
+        if (!target_dir)
+          target_dir = target_dir_owned = g_build_filename (dnf_repo_get_location (src), "/packages/", NULL);
```

Comment:
And the problem here is pretty subtle we reassigning to the input parameter each time in the loop.  So the first time it will be `NULL` e.g., but not the second.

Reply:
Thanks, I had seen the CI error but hadn't dug into it yet.
Fixed!